src/Config/TerminusConfig.php

```diff
@@ -26,46 +26,70 @@ class TerminusConfig extends \Robo\Config
     protected $source_name = 'Unknown';
 
     /**
-     * Set add all the values in the array to this Config object.
-    }
-     * @param array $array
+     * Ensures a directory exists
+     *
+     * @param string $name  The name of the config var
+     * @param string $value The value of the named config var
+     * @return boolean|null
      */
-    public function fromArray(array $array = [])
+    public function ensureDirExists($name, $value)
     {
-        foreach ($array as $key => $val) {
-            $this->set($key, $val);
+        if ((strpos($name, 'TERMINUS_') !== false) && (strpos($name, '_DIR') !== false) && ($value != '~')) {
+            try {
+                $dir_exists = (is_dir($value) || (!file_exists($value) && @mkdir($value, 0777, true)));
+            } catch (\Exception $e) {
+                return false;
+            }
+            return $dir_exists;
         }
+        return null;
     }
 
     /**
-     * Convert the config to an array.
+     * Override the values in this Config object with the given input Config
      *
-     * @return array
+     * @param \Pantheon\Terminus\Config\TerminusConfig $in
      */
-    public function toArray()
+    public function extend(TerminusConfig $in)
     {
-        $out = [];
-        foreach ($this->keys() as $key) {
-            $out[$key] = $this->get($key);
+        foreach ($in->keys() as $key) {
+            $this->set($key, $in->get($key));
+            // Set the source of this variable to make tracking config easier.
+            $this->setSource($key, $in->getSource($key));
         }
-        return $out;
     }
 
     /**
-     * Set a config value. Converts key from Terminus constant (TERMINUS_XXX) if needed.
+     * Ensures that directory paths work in any system
      *
-     * @param string $key
-     * @param mixed $value
+     * @param string $path A path to set the directory separators for
+     * @return string
+     */
+    public function fixDirectorySeparators($path)
+    {
+        return str_replace(['/', '\\',], DIRECTORY_SEPARATOR, $path);
+    }
+
+    /**
+     * Formats given Unix datetimes to the Terminus standard format given by the configuration
      *
-     * @return $this
+     * @param string $datetime A Unix datetime to format
+     * @return string Returns a formatted datetime
      */
-    public function set($key, $value)
+    public function formatDatetime($datetime)
     {
-        // Convert Terminus constant name to internal key.
-        if ($this->keyIsConstant($key)) {
-            $key = $this->getKeyFromConstant($key);
+        return date($this->get('date_format'), (integer)$datetime);
+    }
+
+    /**
+     * Set add all the values in the array to this Config object.
+     * @param array $array
+     */
+    public function fromArray(array $array = [])
```

Comment:
Robo\Config already has `combine()` that does this:

https://github.com/consolidation/config/blob/master/src/ConfigInterface.php#L69

You could call through to keep the name `fromArray()`.